src/common/ExitManager.ts

```diff
@@ -68,6 +70,54 @@ export default class ExitManager extends ContractsBase {
     )
   }
 
+  async buildPayloadForExitHermoine(burnTxHash, logEventSig) {
+    // check checkpoint
+    const lastChildBlock = await this.rootChain.getLastChildBlock()
+    const burnTx = await this.web3Client.getMaticWeb3().eth.getTransaction(burnTxHash)
```

Comment:
query `hermione` instead of `web3` call

Reply:
Does this needs to be done ? 
I don't think this one call will have a substantial difference. 
The best solution is to move this completely to a backend server, create the tx object on the backend and then make the user sign it from front end. 

Denis is already working on improving the exit processing speed.

Reply:
you can get `burnTx` and `txReceipt` in single call if you are querying data from Hermione
I'm not sure exactly what Denis is doing and how it going to be integrated into SDK

but here you are trying optimizing proof by reducing `web3` call then it's better to make changes at all possible places

Reply:
Denis is not working on improving payload build speed - he will just create a stateless modular library for proof creation. So Hermione still needs to be used.

Reply:
@rahuldamodar94